ros/api/v1/recommendations.py

```diff
@@ -44,14 +45,19 @@ class RecommendationsApi(Resource):
 
     @marshal_with(data_fields)
     def get(self, host_id):
+        system = ''
         if not is_valid_uuid(host_id):
             abort(404, message='Invalid host_id, Id should be in form of UUID4')
 
         ident = identity(request)['identity']
 
         filter_description = request.args.get('description')
 
-        system = system_ids_by_org_id(ident['org_id'], True).filter(System.inventory_id == host_id).first()
+        system_query = system_ids_by_org_id(ident['org_id'], True).filter(System.inventory_id == host_id)
+        try:
```

Comment:
Do we require `try..except` block?
We have an `if` block at line 62 which is checking the system's existence.

Reply:
Only try..except is sufficient! I removed the if block!